data-structures/linkedlist-exercises/exercise4.py

```diff
@@ -0,0 +1,64 @@
+# Write a Python program to access a specific item in a singly linked list using index value.
+
+
+class Node:
+    def __init__(self, value, next=None):
+        self.value = value
+        self.next = next
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+        self.tail = None
+        self.count = 0
+
+    def append(self, value):
+        """Add an item to the end of the list."""
+        new_item = Node(value)
+        if not self.head:
+            self.head = new_item
+            self.tail = new_item
+        else:
+            self.tail.next = new_item
+            self.tail = new_item
+        self.count += 1
+
+    def iterate(self):
+        """Print all of the elements in the list"""
+        current = self.head
+        while current:
+            value = current.value
+            current = current.next
+            yield value
```

Comment:
so the yeild function means you continue the loop when you call the function again?